src/Microsoft.TestPlatform.AdapterUtilities/ManagedNameUtilities/ManagedNameHelper.Reflection.cs

```diff
@@ -81,20 +81,50 @@ public static void GetManagedName(MethodBase method, out string managedTypeName,
     /// <see href="https://github.com/microsoft/vstest-docs/blob/main/RFCs/0017-Managed-TestCase-Properties.md">the RFC</see>.
     /// </remarks>
     public static void GetManagedName(MethodBase method, out string managedTypeName, out string managedMethodName, out string[] hierarchyValues)
+    {
+        GetManagedName(method, out managedTypeName, out managedMethodName);
+        GetManagedNameAndHierarchy(method, true, out _, out _, out hierarchyValues);
+    }
+
+    /// <summary>
+    /// Gets default hierarchy values for a given <paramref name="method"/>.
+    /// </summary>
+    /// <param name="method">
+    /// A <see cref="MethodBase" /> instance to get default hierarchy values.
+    /// </param>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="method"/> is null.
+    /// </exception>
+    /// <exception cref="NotSupportedException">
+    /// <paramref name="method"/> must describe a method.
+    /// </exception>
+    /// <exception cref="NotImplementedException">
+    /// Required functionality on <paramref name="method"/> is missing on the current platform.
+    /// </exception>
+    /// <returns>
+    /// The hierarchy values.
+    /// </returns>
+    public static string[] GetManagedHierarchy(MethodBase method)
+    {
+        GetManagedNameAndHierarchy(method, true, out _, out _, out var hierarchyValues);
+
+        return hierarchyValues;
+    }
+
+    private static void GetManagedNameAndHierarchy(MethodBase method, bool useClosedTypes, out string managedTypeName, out string managedMethodName, out string[] hierarchyValues)
```

Comment:
It's not entirely clear to me what is a "closed type", could you please either explain here or add comment on one of the methods?

Reply:
Yes. 'List<string>' is a closed type; and 'List<T>' is open.